src/server/game/Maps/Map.cpp

```diff
@@ -458,8 +458,13 @@ bool Map::AddPlayerToMap(Player* player, bool initPlayer /*= true*/)
     PhasingHandler::SendToPlayer(player);
 
     if (player->IsAlive())
+    {
         ConvertCorpseToBones(player->GetGUID());
 
+        if (Instanceable())
+            player->RemoveAurasWithInterruptFlags(SpellAuraInterruptFlags2::EnteringInstance);
```

Comment:
Not sure if we should check IsAlive() for this, might as well leave it outside of the check since we don't even do that in Battleground::RemovePlayerAtLeave to remove battleground-specific auras.